# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 48eaa84c206d
# Full Hash: 48eaa84c206dbdda637c57b55ddbd07e975ba103
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-23 16:15:06
# Regressor Bug: 1637500
# File Overlap Count: 2
# Description:
#   Bug 1637500 - P2. Rename methods as they are not always dealing with "threads". r=froydnj
#   
#   Before P1, GetCurrentThreadSerialEventTarget would have always returned the same data as NS_GetCurrentThread, making the comment incorrect Now it will properly return the running TaskQueue if any.
#   
#   This change of name more clearly exposes what they are doing, as we aren't always dealing with threads directly; but a nsISerialEventTarget
# ==============================================================================

diff -r 00687d06a8a6 -r 48eaa84c206d dom/base/Document.cpp
--- a/dom/base/Document.cpp	Tue Jun 23 05:02:39 2020 +0000
+++ b/dom/base/Document.cpp	Tue Jun 23 05:05:36 2020 +0000
@@ -1523,7 +1523,7 @@
     ContentChild* cc = ContentChild::GetSingleton();
     MOZ_ASSERT(cc);
     cc->SendAddCertException(certSerialized, flags, host, port, aIsTemporary)
-        ->Then(GetCurrentThreadSerialEventTarget(), __func__,
+        ->Then(GetCurrentSerialEventTarget(), __func__,
                [promise](const mozilla::MozPromise<
                          nsresult, mozilla::ipc::ResponseRejectReason,
                          true>::ResolveOrRejectValue& aValue) {
@@ -15769,7 +15769,7 @@
   }
 
   return wgc->SendGetContentBlockingEvents()->Then(
-      GetCurrentThreadSerialEventTarget(), __func__,
+      GetCurrentSerialEventTarget(), __func__,
       [](const WindowGlobalChild::GetContentBlockingEventsPromise::
              ResolveOrRejectValue& aValue) {
         if (aValue.IsResolve()) {
@@ -15929,7 +15929,7 @@
         }
 
         self->AutomaticStorageAccessPermissionCanBeGranted()->Then(
-            GetCurrentThreadSerialEventTarget(), __func__,
+            GetCurrentSerialEventTarget(), __func__,
             [p, pr, sapr,
              inner](const AutomaticStorageAccessPermissionGrantPromise::
                         ResolveOrRejectValue& aValue) -> void {
@@ -15964,7 +15964,7 @@
                     p->Resolve(choice, __func__);
                   } else {
                     sapr->MaybeDelayAutomaticGrants()->Then(
-                        GetCurrentThreadSerialEventTarget(), __func__,
+                        GetCurrentSerialEventTarget(), __func__,
                         [p, choice] { p->Resolve(choice, __func__); },
                         [p] { p->Reject(false, __func__); });
                   }
@@ -15985,7 +15985,7 @@
           NodePrincipal(), bc, ContentBlockingNotifier::eStorageAccessAPI,
           performFinalChecks)
           ->Then(
-              GetCurrentThreadSerialEventTarget(), __func__,
+              GetCurrentSerialEventTarget(), __func__,
               [outer, promise] {
                 // Step 10. Grant the document access to cookies and store
                 // that fact for
@@ -16020,7 +16020,7 @@
     return cc
         ->SendAutomaticStorageAccessPermissionCanBeGranted(
             IPC::Principal(NodePrincipal()))
-        ->Then(GetCurrentThreadSerialEventTarget(), __func__,
+        ->Then(GetCurrentSerialEventTarget(), __func__,
                [](const ContentChild::
                       AutomaticStorageAccessPermissionCanBeGrantedPromise::
                           ResolveOrRejectValue& aValue) {